File: 387. First Unique Character in a String/Approach03.py
from typing import *
import collections
import sys

class Solution:
    def firstUniqChar(self, s: str) -> int:
        f = collections.Counter(s)
        result = sys.maxsize
        
        filter_freq = [ ord(k) - ord('a') for k,v in f.items() if v==1]  
        n = len(filter_freq)
        
        for i in range(n):            
                index = s.index(chr(filter_freq[i] + ord('a')))
                result = min(index, result)
        if(result == sys.maxsize):
            return -1
        return result

File: 387. First Unique Character in a String/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> mp;
        for(auto& ch: s){
            mp[ch]++;
        }
        
        int n = s.length();
        for(int i=0; i<n; i++){
            if(mp[s[i]] == 1){
                return i;
            }
        }
        return -1;
    }
};
